Add yue language support to Google Translate TTS

beginner2047 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
No

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Change the gTTS to 2.5.3 in order to support yue language in Google Translate TTS [https://github.com/home-assistant/core/issues/130280](https://github.com/home-assistant/core/issues/130280)

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #130280 
- Link to documentation pull request: [google_translate documentation](https://www.home-assistant.io/integrations/google_translate)
[google_translate source](https://github.com/home-assistant/core/tree/dev/homeassistant/components/google_translate)
A link to the release notes of this package version, and all versions in between.
https://github.com/pndurette/gTTS/releases
A link to the changelog of this package.
https://github.com/pndurette/gTTS/blob/master/CHANGELOG.md
A link to a Git(Hub) diff/compare view from the current version to the bumped version.
https://github.com/pndurette/gTTS/compare/v2.2.4...v2.5.3
## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3-m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

beginner2047 committed:
Update const.py

Add yue

beginner2047 committed:
Change gTTS to 2.5.3 which support yue

beginner2047 committed:
Merge branch 'home-assistant:dev' into dev

beginner2047 commented:
Here is my test screen locally
![image](https://github.com/user-attachments/assets/b29f44c8-5c24-4966-ac3f-a55940dad72a)

![image](https://github.com/user-attachments/assets/b51011ac-2718-43b7-a418-49c15266d838)

![image](https://github.com/user-attachments/assets/3b73611a-f3f4-41c4-9444-0708b53aa596)

![image](https://github.com/user-attachments/assets/2f2df4f8-eb11-4bb3-b9e5-227c14f40f1e)

beginner2047 committed:
Merge branch 'dev' into dev

joostlek reviewed (changes requested):
Thanks for bumping the dependency! However, could you please update the PR description to contain at least one (or multiple) of the following:

- A link to the release notes of this package version, and all versions in between.
- A link to the changelog of this package.
- A link to a Git(Hub) diff/compare view from the current version to the bumped version.

This allows us to review upstream changes, which is needed to decide this change is working as intended and/or if we can include it in, for example, a patch release of Home Assistant.

Thanks already! 👍

Also, please run `gen_requirements_all.py` as the updated requirements need to be updated in the requirement files

beginner2047 committed:
Merge branch 'home-assistant:dev' into dev

beginner2047 commented:
A link to the release notes of this package version, and all versions in between.
[https://github.com/pndurette/gTTS/releases](url)
A link to the changelog of this package.
[https://github.com/pndurette/gTTS/blob/master/CHANGELOG.md](url)
A link to a Git(Hub) diff/compare view from the current version to the bumped version.
[https://github.com/pndurette/gTTS/compare/v2.2.4...v2.5.3](url)

beginner2047 committed:
Update requirements after running gen_requirements_all.py

beginner2047 commented:
python3 script/gen_requirements_all.py

git status
On branch dev
Your branch is up to date with 'origin/dev'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   homeassistant/components/google_translate/const.py
        modified:   homeassistant/components/google_translate/manifest.json
        modified:   requirements_all.txt
        modified:   requirements_test_all.txt

 git diff
diff --git a/homeassistant/components/google_translate/const.py b/homeassistant/components/google_translate/const.py
index ed9709d281..ab0291bc58 100644
--- a/homeassistant/components/google_translate/const.py
+++ b/homeassistant/components/google_translate/const.py
@@ -88,6 +88,7 @@ SUPPORT_LANGUAGES = [
     "uk",
     "ur",
     "vi",
+    "yue",
     # dialects
     "zh-CN",
     "zh-cn",
diff --git a/homeassistant/components/google_translate/manifest.json b/homeassistant/components/google_translate/manifest.json
index 7074d0ed44..b5b1f67067 100644
--- a/homeassistant/components/google_translate/manifest.json
+++ b/homeassistant/components/google_translate/manifest.json
@@ -6,5 +6,5 @@
   "documentation": "https://www.home-assistant.io/integrations/google_translate",
   "iot_class": "cloud_push",
   "loggers": ["gtts"],
-  "requirements": ["gTTS==2.2.4"]
+  "requirements": ["gTTS==2.5.3"]
 }
diff --git a/requirements_all.txt b/requirements_all.txt
index cc17cbde00..95a8046aba 100644
--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -959,7 +959,7 @@ fritzconnection[qr]==1.14.0
 fyta_cli==0.7.0

 # homeassistant.components.google_translate
-gTTS==2.2.4
+gTTS==2.5.3

 # homeassistant.components.gardena_bluetooth
 gardena-bluetooth==1.5.0
diff --git a/requirements_test_all.txt b/requirements_test_all.txt
index bc12b9b634..6b6d296925 100644
--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -812,7 +812,7 @@ fritzconnection[qr]==1.14.0
 fyta_cli==0.7.0

 # homeassistant.components.google_translate
-gTTS==2.2.4
+gTTS==2.5.3

 # homeassistant.components.gardena_bluetooth
 gardena-bluetooth==1.5.0

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev

joostlek commented:
You should commit those `requirements.txt` and `requirements_test.txt` and push them

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Update requirements after running gen_requirements_all.py

beginner2047 commented:
May I know if you are referring to requirements**_all**.txt and requirements_test**_all**.txt. These have been updated and pushed. If you meant different files, please let me know. 

After running gen_requirements_all.py, only requirements**_all**.txt and requirements_test**_all**.txt are changed.

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev

beginner2047 committed:
Merge branch 'dev' into dev
